extensions/amp-analytics/0.1/test/test-amp-analytics.js

```diff
@@ -902,21 +902,30 @@ describe('amp-analytics', function() {
   describe('expandTemplate_', () => {
     const vars = {
       'vars': {'1': '1${2}', '2': '2${3}', '3': '3${4}', '4': '4${1}'}};
+    let analytics;
+
+    beforeEach(() => {
+      analytics = getAnalyticsTag(trivialConfig);
+    });
 
     it('expands nested vars', () => {
-      const analytics = getAnalyticsTag(trivialConfig);
       const actual = analytics.expandTemplate_('${1}', vars);
       expect(actual).to.equal('123%252524%25257B4%25257D');
     });
 
     it('limits the recursion to n', () => {
-      const analytics = getAnalyticsTag(trivialConfig);
       let actual = analytics.expandTemplate_('${1}', vars, {}, 3);
       expect(actual).to.equal('1234%25252524%2525257B1%2525257D');
 
       actual = analytics.expandTemplate_('${1}', vars, {}, 5);
       expect(actual).to.equal('123412%252525252524%25252525257B3%25252525257D');
     });
+
+    it('works with params', () => {
```

Comment:
I wasn't sure if you wanted me to add a unittest for `getNameArgs()` or something else. I added two tests. the one that goes through the whole code flow is not a unittest as it mocks parts of url replacement. 

(getting actual query params is difficult here because the win.loc is about:srcdoc or something - no query params.)